### PR TITLE
Use generic list in ListView

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Use generic list in ListView.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6398)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ListView.cs [hunk 1]
```diff
@@ -136,7 +136,7 @@ public partial class ListView : Control
         // We do not try to sort listItemsArray as items are added, but during a handle recreate
         // we will make sure we get the items in the same order the ListView displays them.
         private readonly Hashtable listItemsTable = new Hashtable(); // elements are ListViewItem's
-        private ArrayList listItemsArray = new ArrayList(); // elements are ListViewItem's
+        private List<ListViewItem> _listViewItems = new List<ListViewItem>();
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListView.cs:139]:
```suggestion
        private List<ListViewItem> _listViewItems = new ();
```

Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/ListView.cs:139]: and should this nullable?

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ListView.cs]: > and should this nullable?

Nullability is not enabled in this file.

**Review by WPMGPRoSToTeMa — APPROVED** [review]
🚀

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**